src/crypto/store/indexeddb-crypto-store-backend.js

```diff
@@ -203,6 +204,24 @@ export class Backend {
         return promiseifyTxn(txn).then(() => result);
     }
 
+    /**
+     *
+     * @param {Number} wantedState
+     * @return {Promise<Array<*>>} All elements in a given state
+     */
+    getAllOutgoingRoomKeyRequestsByState(wantedState) {
+        return new Promise((resolve, reject) => {
+            const txn = this._db.transaction("outgoingRoomKeyRequests", "readonly");
+            const store = txn.objectStore("outgoingRoomKeyRequests");
+            const index = store.index("state");
+            const request = index.getAll(wantedState);
+
+            request.onsuccess = (ev) => resolve(ev.target.result);
+            request.onerror = (ev) => reject(ev.target.error);
+            request.onabort = (ev) => reject(ev.target.error);
```

Comment:
`onabort` doesn't seem to be a thing on an `IDBRequest` according to MDN. I think an aborted txn rather uses onerror.

Reply:
while I was testing I ran into a situation where it fired, but I can take this out